src/vulkan-renderer/world/cube.cpp

```diff
@@ -426,4 +424,67 @@ std::vector<PolygonCache> Cube::polygons(const bool update_invalid) const {
     collect(*this);
     return polygons;
 }
+std::optional<std::shared_ptr<Cube>> Cube::neighbor(const NeighborAxis axis, const NeighborDirection direction) {
+    assert(!is_root());
+
+    // Each axis only requires information and manipulation of one (relevant) bit to find the neighbor.
+    const auto relevant_bit_index = static_cast<std::uint8_t>(axis); // bit index of the axis we are working on
```

Comment:
Can't you use a bitset?

Reply:
How? Simpy changing the type used in axis and here would result in a type mismatch in the `get_bit` and `toggle_bit` methods below, wouldn't it?

Reply:
Also, it would require to include and use bitset just for this two methods which I think might be a little heavy.